packages/create-amplify/src/package-manager-controller/package_manager_controller_factory.ts

```diff
@@ -0,0 +1,157 @@
+import { existsSync as _existsSync } from 'fs';
+import { execa as _execa } from 'execa';
+import * as path from 'path';
+import {
+  type PackageManagerName,
+  type PackageManagers,
+} from './package_manager.js';
+import { logger } from '../logger.js';
+import { NpmPackageManagerController } from './npm_package_manager_controller.js';
+import { PnpmPackageManagerController } from './pnpm_package_manager_controller.js';
+import { YarnClassicPackageManagerController } from './yarn_classic_package_manager_controller.js';
+import { YarnModernPackageManagerController } from './yarn_modern_package_manager_controller.js';
+import {
+  NpmProjectInitializer,
+  PnpmProjectInitializer,
+  YarnClassicProjectInitializer,
+  YarnModernProjectInitializer,
+} from '../project-initializer/index.js';
+
+export type DependencyType = 'dev' | 'prod';
+
+/**
+ *
+ */
+export abstract class PackageManagerController {
+  abstract installDependencies: (
+    packageNames: string[],
+    type: DependencyType
+  ) => Promise<void>;
+}
+
+/**
+ * packageManagerControllerFactory
+ */
+export class PackageManagerControllerFactory {
+  private readonly existsSync = _existsSync;
+  packageManagerName: PackageManagerName;
+  private readonly packageManagers: PackageManagers = {
```

Comment:
This should all be internal to the controllers. Also, it looks like these values are also duplicated in the controllers. How will we keep it in sync?

Reply:
removed